pkg/api/validate/dynamic/workersku.go

```diff
@@ -0,0 +1,55 @@
+package dynamic
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/computeskus"
+)
+
+// ValidateWorkerSku uses resourceSkusClient to ensure that the VMSize listed in the cluster document WorkerProfile is available for use in the target region.
+func (dv *dynamic) ValidateWorkerSku(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster) error {
```

Comment:
Do we only care about workers or should we check masters too?

Reply:
Good question, I think we should also check masters. Recently we had an RP e2e failure during rollout because Standard_D8s_v2 was restricted in one of the zones where it was trying to deploy, so we know that it's also possible to have sku restrictions that affect masters.

Reply:
@m1kola I have added validation / unit test coverage for masters as well. Let me know what you think.
![nov_30th_sku_code_coverage](https://user-images.githubusercontent.com/56140267/144132027-1bc5f618-516e-44aa-8006-f45f3a1ed1b3.png)